crates/node/src/command.rs

```diff
@@ -10,6 +15,30 @@ use crate::constants::DEV_CHAIN_ID;
 use crate::constants::SHARINGAN_CHAIN_ID;
 use crate::{chain_spec, service};
 
+fn spit_spec_to_json(base_config: impl ChainSpec) -> Result<Box<dyn ChainSpec>, String> {
```

Comment:
Not a big fan of this solution of writing to disk to read it again just after.
Is there a way to fix the solution in a more elegant way?
Did you manage to understand why there was incoherences between the two nodes?

Reply:
The plan was to split the issue into two different PRs as trying to find the original bug is taking longer than I originally thought that it would. so this PR fixes the initial issue #1468 while this PR #1498 will provide a better solution to this issue #1485